crates/exec-wasmtime/src/lib.rs

```diff
@@ -51,10 +68,36 @@ pub fn execute() -> anyhow::Result<()> {
     // The FD is managed by the host or its parent.
     forget(host);
 
-    let args = toml::from_str::<Args>(&args).context("failed to decode arguments")?;
-
-    execute_with_args(args)?;
-
+    let Args {
+        log_level,
+        #[cfg(feature = "bench")]
+        profile,
+        package,
+    } = toml::from_str(&args).context("failed to decode arguments")?;
+
+    #[cfg(feature = "bench")]
+    let (flame_layer, _guard) = if let Some(profile) = profile {
```

Comment:
I would prefer `let foo = match bar {` over `let foo = if let Some(foo) = bar {` in cases like this where you handle all the variants anyway.

Reply:
I disagree with this, because that would add an unnecessary indentation level

Reply:
My criterion for when to use `match` over `if let` is whenever I'm handling all of the variants of an enum. I find it to be more readable, despite the extra indentation level. Do you not use `match` at all?

Reply:
I use `match` when there are more than two cases to match. Matching an `Option` using a `match` is redundant in my opinion. That's also why multitude of helpers exist for handling this (`.map*`, `.unwrap*` etc).

Reply:
There are also plenty of `let x = if let Some(..) = {} else {}` examples in our codebase

Reply:
```
$ rg 'if let Some\('
tests/integration.rs
151:    if let Some(stdin) = stdin {
244:    if let Some(expected_stdout) = expected_stdout {
256:    if let Some(expected_stderr) = expected_stderr {

tests/client/util.rs
88:    if let Some(msg) = failed_test {

tests/crates/enarx_syscall_tests/src/lib.rs
281:    let arg1 = if let Some(ref nonce) = nonce {
287:    let arg0 = if let Some(nonce) = nonce {
293:    let arg3 = if let Some(ref buf) = buf {
299:    let arg2 = if let Some(buf) = buf {

tests/crates/enarx_exec_tests/src/bin/sgx_attestation.rs
47:    let nonce_ptr = if let Some(ref nonce) = nonce {
53:    let nonce_len = if let Some(ref nonce) = nonce {
59:    let buf_len = if let Some(ref buf) = buf {
65:    let buf_ptr = if let Some(buf) = buf {

tests/crates/enarx_exec_tests/src/bin/sev_attestation.rs
134:    let arg1 = if let Some(ref nonce) = nonce {
140:    let arg0 = if let Some(nonce) = nonce {
146:    let arg3 = if let Some(ref buf) = buf {
152:    let arg2 = if let Some(buf) = buf {

tests/wasm/mod.rs
37:            if let Some(conf) = conf {

crates/shim-kvm/src/hostcall.rs
688:    if let Some(mut host_call) = HostCall::try_new(SHIM_LOCAL_STORAGE.write().deref_mut()) {

crates/shim-kvm/src/debug.rs
133:        if let Some(rip_rbp) = rbp.checked_add(size_of::<usize>() as _) {
138:            if let Some(rip) = rip
202:        if let Some(rip_rbp) = rbp.checked_add(size_of::<usize>() as _) {
211:            if let Some(rip) = rip.checked_sub(1) {
215:                if let Some(rip) = rip.checked_sub(shim_offset) {
219:                    if let Some(rip) = rip.checked_sub(EXEC_VIRT_ADDR.read().as_u64() as _) {

crates/shim-sgx/src/handler/mod.rs
345:        if let Some(addr) = addr {
472:                if let Some(block) = block {

crates/exec-wasmtime/src/workload.rs
89:    let entry = if let Some(entry) = dir.get(&PACKAGE_CONFIG) {
211:                let config = if let Some(conf) = conf.as_mut() {

crates/exec-wasmtime/src/lib.rs
79:    let (flame_layer, _guard) = if let Some(profile) = profile {

crates/exec-wasmtime/src/runtime/mod.rs
53:        let certs = if let Some(url) = steward {

crates/sallyport/src/item/block.rs
20:        let item_size = if let Some(item_size) = item_count.checked_mul(LARGEST_ITEM_SIZE) {
26:        let size = if let Some(size) = item_size.checked_add(data_size) {

crates/sallyport/src/guest/handler.rs
580:        if let Some(oldact) = oldact {
583:        if let Some(act) = act {
982:                    if let Some(oldact) = oldact {

crates/sallyport/src/guest/call/syscall/ioctl.rs
54:        let argp = if let Some(argp) = self.0.argp {

crates/sallyport/src/guest/call/syscall/nanosleep.rs
42:        let (rem, rem_offset) = if let Some(rem) = self.rem {

crates/sallyport/src/guest/call/enarxcall/park.rs
29:        let (timeout, timeout_offset) = if let Some(timeout) = self.timeout {

crates/sallyport/src/guest/call/syscall/clock_getres.rs
27:        let (res, res_offset) = if let Some(res) = self.res {

src/drawbridge.rs
140:    if let Some(token) = provided_token {
142:    } else if let Some(helper) = helper {
153:        } else if let Some(code) = output.status.code() {
182:    if let Some(ca_bundle_path) = ca_bundle {
232:    if let Some(helper) = helper {
252:            if let Some(code) = output.code() {

src/backend/kvm/thread.rs
163:                            if let Some(Item::Enarxcall(enarxcall, data)) =

src/cli/sign.rs
140:        let binary = if let Some(ref path) = self.binpath {
155:            let exec = if let Some(ref e) = binary {
157:            } else if let Some(e) = EXECS.iter().find(|w| w.with_backend(backend)) {
185:        if let Some(path) = self.out {

src/exec/mod.rs
118:    if let Some(conf) = conf {

src/backend/sev/snp/launch/linux.rs
138:            ma_uaddr: if let Some(addr) = start.ma_uaddr {

src/cli/mod.rs
73:        let (flame_layer, _guard, profile) = if let Some(ref profile) = self.logger.profile {
172:        if let Some(ref name) = self.backend {

src/cli/key/sgx/create.rs
31:        if let Some(path) = self.out {

src/cli/key/sgx/digest.rs
52:        if let Some(path) = self.out {

src/backend/sgx/enarxcall.rs
45:                    if let Some(thread) = keep.clone().spawn()? {
59:                if let Some(mut thread) = thread {

src/cli/key/sev/create.rs
30:        if let Some(path) = self.out {

src/backend/sev/hasher.rs
42:        if let Some(src) = src {

src/cli/platform/info.rs
120:                if let Some(ref info) = datum.info {
127:                if let Some(mesg) = datum.mesg.as_ref() {

src/backend/sgx/builder.rs
135:        let signature = if let Some(signatures) = builder.cnfg.signatures {

src/cli/key/sev/digest.rs
65:        if let Some(path) = self.out {

src/backend/sev/builder.rs
206:        let finish = if let Some(signatures) = signatures {
```